Sources/Brave/Frontend/Browser/Playlist/Controllers/PlaylistViewController.swift

```diff
@@ -384,6 +384,26 @@ class PlaylistViewController: UIViewController {
         self.playerView.controlsView.repeatButton.setImage(UIImage(named: "playlist_repeat_all", in: .module, compatibleWith: nil)!, for: .normal)
       }
     }.store(in: &playerStateObservers)
+    
+    player.publisher(for: .previousTrack).sink { [weak self] _ in
+      guard let self = self, !PlaylistCarplayManager.shared.isCarPlayAvailable else { return }
+      self.onPreviousTrack(self.playerView, isUserInitiated: true)
+    }.store(in: &playerStateObservers)
+
+    player.publisher(for: .nextTrack).sink { [weak self] _ in
+      guard let self = self, !PlaylistCarplayManager.shared.isCarPlayAvailable else { return }
+      self.onNextTrack(self.playerView, isUserInitiated: true)
+    }.store(in: &playerStateObservers)
+    
+    player.publisher(for: .seekBackward).sink { [weak self] _ in
+      guard let self = self, !PlaylistCarplayManager.shared.isCarPlayAvailable else { return }
+      self.seekBackwards(self.playerView)
+    }.store(in: &playerStateObservers)
+    
+    player.publisher(for: .seekForward).sink { [weak self] _ in
+      guard let self = self, !PlaylistCarplayManager.shared.isCarPlayAvailable else { return }
```

Comment:
Why are these `!PlaylistCarplayManager.shared.isCarPlayAvailable` shouldn't it be the opposite? Don't you need CarPlay to be available for seek forwards/backwards, etc to work? Or is this somehow ignoring the action if CarPlay is active because some other mechanism takes over? If this is the case, maybe a comment here would be useful because its not at all intuitive.

Reply:
This is the playlist controller. CarPlay is a separate controller and handles it's own stuff.

These functions are not for CarPlay. So there's a check to make sure that the user isn't using the phone and CarPlay at the same time. If they are, the CarPlay one takes precedence. There is a comment in the .finishedPlaying.